server-spi-private/src/main/java/org/keycloak/models/utils/RepresentationToModel.java

```diff
@@ -1703,7 +1704,7 @@ private static void updateOrganizationBroker(IdentityProviderRepresentation repr
 
             if (StringUtil.isBlank(domain)) {
                 representation.getConfig().remove(OrganizationModel.ORGANIZATION_DOMAIN_ATTRIBUTE);
-            } else if (org.getDomains().map(OrganizationDomainModel::getName).noneMatch(domain::equals)) {
+            } else if (!ANY_DOMAIN.equalsIgnoreCase(domain) && org.getDomains().map(OrganizationDomainModel::getName).noneMatch(domain::equals)) {
```

Comment:
I am myself not a friend of `equalsIgnoreCase` here. This would allow `aNy`, `any`, `ANY` etc. to be sent through the rest api.
From a database point of view, always using `ANY` is easier to handle, espacially if later on any migrations need to be done, or other features need to access the domains of an organization.

I would therefore strongly recommend to only input `ANY` into the database. Either by validating and rejecting invalid entries, or by correcting an `any` automatically to `ANY`.

Reply:
Yeah, I was not totally happy `equalsIgnoreCase` either. I think your idea to add a validation when converting the representation to model is the best to go to ensure the DB will always store `ANY` instead of variations of that word.